### PR TITLE
Bug/18436 entity unregister redirect

### LINKED ISSUE #18436: 🐛 Bug Report: Post unregistering an entity, user gets redirected to "/" instead of "/catalog"
### 📜 Description

Post unregistering an entity, user gets redirected to "/" instead of "/catalog"

### 👍 Expected behavior

It should get redirected to "/catalog"

### 👎 Actual Behavior with Screenshots

Post unregistering it got redirected to default route "/".

### 👟 Reproduction steps

1. Go To the Catalog Page.
2. Go to entity details Page.
3. Unregister the entity.

### 📃 Provide the context for the Bug.

_No response_

### 🖥️ Your Environment

_No response_

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

### PR DESCRIPTION
**Pull request by NeuralEvolution**
## Hey, I just made a Pull Request!

fixes #18436 navigates to catalog root using rootRouteRef after successfully unregistering an entity.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by benjdlambert** [comment]
Hey @NeuralEvolution! Thanks for the contribution!

There's a little bit more context here that I think you might have missed from a previous attempt at this work, but I think it would be good if we used an `externalRouteRef`here instead.

https://github.com/backstage/backstage/pull/18894#issuecomment-1714125186

**Comment by mprinold** [comment]
@benjdlambert Updated the solution to include the external route ref and fall back to the catalog root route ref if no external one is bound. 

After those commits i rebased the fork and this branch as it was behind original and now i'm seeing failing checks. Could do with a bit of guidance on this please.

**Comment by benjdlambert** [comment]
I think that these are failing because of a bad yarn lock because of the addition to `package.json`.

Quick look through the PR though you might not need that dependency at all. As it uses `useNavigate` from `react-router-dom` so maybe mock that instead?

https://github.com/backstage/backstage/pull/20786/files#diff-2b97bcc2e16b1532a23a6719b7a254379e2497905ce705e57685650d74b577fcR52

**Comment by NeuralEvolution** [comment]
@benjdlambert yeah i originally was going to mock useNavigate from react-router-dom however there are other tests in the file that rely on it so that wasn't an option and it seems you cant spy on it either as you get an error "Cannot redefine property useNavigate" which is why i went down the approach of spying on useNavigate from react-router.

**Comment by NeuralEvolution** [comment]
@benjdlambert all checks passing again now

File: plugins/catalog/src/routes.ts [hunk 1]
```diff
@@ -36,6 +36,11 @@ export const createFromTemplateRouteRef = createExternalRouteRef({
   params: ['namespace', 'templateName'],
 });
 
+export const unregisterRedirectRouteRef = createExternalRouteRef({
+  id: 'unregister-redirect',
```
Comment (benjdlambert) [inline plugins/catalog/src/routes.ts:40]: we normally try namespace these. something like `catalog:unregister-redirect` or something works.

File: .changeset/lucky-kids-cough.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-catalog': patch
+---
+
+fixes bug where after unregistering an entity you are redirected to "/". Adds an optional external route "unregisterRedirect" into the catalog plugin which when bound will be navigated to on successful removal of an entity. If the external route is not bound it will default to using the catalog rootRouteRef "/catalog".
```
Comment (benjdlambert) [inline .changeset/lucky-kids-cough.md:5]:
```suggestion
- Fixes bug where after unregistering an entity you are redirected to `/`.
- Adds an optional external route `unregisterRedirect` to override this behaviour to another route.
```

**Review by benjdlambert — CHANGES_REQUESTED** [review]
Nice thanks for this :tada: Think that this is looking pretty good just a few nits.

**Comment by NeuralEvolution** [comment]
@benjdlambert requested changes made, rebased with master branch and all checks passing again.

**Pull request merged**